cloud/scope/object_storage_bucket.go

```diff
@@ -161,6 +162,50 @@ func (s *ObjectStorageBucketScope) ApplyAccessKeySecret(ctx context.Context, key
 	return nil
 }
 
+// getAccessKeysFromSecret gets the access keys id for the OBJ buckets from a secret
+func (s *ObjectStorageBucketScope) GetAccessKeysFromSecret(ctx context.Context, secretName string, secretNamespace string) ([2]float64, error) {
+
+	var newKeys [2]float64
+	// Delete the access keys.
+	objkey := client.ObjectKey{
+		Namespace: secretNamespace,
+		Name:      secretName,
+	}
+	var secret corev1.Secret
+	if err := s.client.Get(ctx, objkey, &secret); err != nil {
+		if apierrors.IsNotFound(err); err != nil {
+			return newKeys, err
+		}
+	}
+	for i, e := range []struct {
+		permission string
+		suffix     string
+	}{
```

Comment:
Since we only need `permission`, this can just loop over `[]string` versus `[]struct{}